codex/purchasing/states/submitted.nim

```diff
@@ -34,7 +34,7 @@ method run*(state: PurchaseSubmitted, machine: Machine): Future[?State] {.async.
     await subscription.unsubscribe()
 
   proc withTimeout(future: Future[void]) {.async.} =
-    let expiry = request.expiry.truncate(int64) + 1
+    let expiry = (await market.getRequestExpiry(request.id)) + 1
```

Comment:
This value should already be on the `purchase.request`,  so no need to go to the chain for this.

Reply:
Actually, this makes me think of [my earlier point](https://github.com/codex-storage/codex-contracts-eth/pull/99/files#r1575703499), where we shouldn't be using expiry as a duration here, simply because the node's `now` is going to be different from the chain's `now`, meaning the time we wait in this state won't be the same as the event that occurs on chain. So I think a better option would be to:
> We could probably update the `purchasing/states/pending` state to listen for the `StorageRequested` event before going to the `submitted` state, passing in the `expiresAt` value from the event to the `submitted` state. In the `submitted` state, we can use `await clock.waitUntil(expiresAt)` and cancel that future if the request is fulfilled.

That will allow to wait for the right time from the chain.